src/foam/nanos/notification/EmailSetting.js

```diff
@@ -11,14 +11,47 @@ foam.CLASS({
   extends: 'foam.nanos.notification.NotificationSetting',
 
   javaImports: [
+    'foam.core.PropertyInfo',
+    'foam.dao.DAO',
+    'foam.nanos.auth.User',
     'foam.nanos.logger.Logger',
     'foam.nanos.notification.email.EmailMessage',
     'foam.util.Emails.EmailsUtility',
     'java.util.Arrays',
+    'java.util.Iterator',
     'java.util.List',
+    'java.util.Map',
+    'static foam.mlang.MLang.EQ'
   ],
 
   methods: [
+    {
+      name: 'mapNotificationArguments',
+      type: 'Map',
+      documentation: `Iterate through arguments to replace propertyInfo values with the notified user' values.`,
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'arguments', type: 'Map' },
+        { name: 'user', type: 'User' }
+      ],
+      javaCode: `
+        Logger logger = (Logger) x.get("logger");
+
+        Iterator entries = arguments.entrySet().iterator();
+        while (entries.hasNext()) {
+          Map.Entry entry = (Map.Entry) entries.next();
+          if ( entry.getValue() instanceof PropertyInfo ) {
+            if ( ! ((PropertyInfo) entry.getValue()).getClassInfo().equals(User.getOwnClassInfo()) ) {
+              logger.error("Cannot set a none User PropertyInfo as an email argument value");
+              continue;
+            }
+            String newValue = (String) ((PropertyInfo) entry.getValue()).get(user);
+            arguments.replace(entry.getKey(), newValue);
```

Comment:
I wonder if setValue would work.
```suggestion
            entry.setValue(newValue);
```
If it does, we might be able to use for-each at line 40 as well.

Reply:
For each enforces generics too strictly for this. However, entry.setValue() does the same as replace so I'll go ahead and change that.

In regards to iterating over maps I found this to be useful https://www.testingexcellence.com/4-different-ways-iterate-map-java/